File: Utils/DataLoader.cs
// Example: utils/DataLoader.cs
using System;
using System.Collections.Generic;
using System.IO;

public class DataLoader
{
    public static List<Dictionary<string, string>> LoadData(string filePath)
    {
        var data = new List<Dictionary<string, string>>();
        var lines = File.ReadAllLines(filePath);

        if (lines.Length > 0)
        {
            var headers = lines[0].Split(',');

            for (int i = 1; i < lines.Length; i++)
            {
                var values = lines[i].Split(',');
                var row = new Dictionary<string, string>();

                for (int j = 0; j < headers.Length; j++)
                {
                    row[headers[j]] = values[j];
                }

                data.Add(row);
            }
        }

        return data;
    }

    public static List<Dictionary<string, string>> PreprocessData(List<Dictionary<string, string>> data)
    {
        // Example preprocessing steps
        data.RemoveAll(row => row.Values.Contains(null));
        return data;
    }

    public static void Main(string[] args)
    {
        string filePath = "path/to/data.csv";
        var data = LoadData(filePath);
        var preprocessedData = PreprocessData(data);

        foreach (var row in preprocessedData)
        {
            foreach (var kvp in row)
            {
                Console.Write($"{kvp.Key}: {kvp.Value} ");
            }
            Console.WriteLine();
        }
    }
}

File: Utils/data_loader.py
# Example: utils/data_loader.py
import os
import pandas as pd

def load_data(data_dir, file_name):
    """
    Load data from a CSV file.

    Parameters:
    data_dir (str): Directory where the data file is located.
    file_name (str): Name of the data file.

    Returns:
    pd.DataFrame: Loaded data as a pandas DataFrame.
    """
    file_path = os.path.join(data_dir, file_name)
    data = pd.read_csv(file_path)
    return data

def preprocess_data(data):
    """
    Preprocess the data.

    Parameters:
    data (pd.DataFrame): Data to preprocess.

    Returns:
    pd.DataFrame: Preprocessed data.
    """
    # Example preprocessing steps
    data = data.dropna()  # Remove missing values
    data = data.reset_index(drop=True)  # Reset index
    return data

# Example usage
if __name__ == "__main__":
    data_dir = "path/to/data"
    file_name = "data.csv"
    data = load_data(data_dir, file_name)
    preprocessed_data = preprocess_data(data)
    print(preprocessed_data.head())
